src/Symfony/Contracts/HttpClient/HttpClientInterface.php

```diff
@@ -53,6 +53,7 @@ interface HttpClientInterface
         'proxy' => null,        // string - by default, the proxy-related env vars handled by curl SHOULD be honored
         'no_proxy' => null,     // string - a comma separated list of hosts that do not require a proxy to be reached
         'timeout' => null,      // float - the inactivity timeout - defaults to ini_get('default_socket_timeout')
+        'max_duration' => null, // float - the maximum execution time of the whole request (including the connection time)
```

Comment:
for the request+response as a whole ~(including the connection time)~

Reply:
max_duration => 0  too isn't it?

Reply:
sure, let's do